iothub/device/src/InternalClient.cs

```diff
@@ -1238,6 +1242,11 @@ public Task UpdateReportedPropertiesAsync(TwinCollection reportedProperties, Can
         //  Codes_SRS_DEVICECLIENT_18_005: When a patch is received from the service, the `callback` shall be called.
         internal void OnReportedStatePatchReceived(TwinCollection patch)
         {
+            if (this.desiredPropertyUpdateCallback == null)
```

Comment:
`this.desiredPropertyUpdateCallback?.Invoke(patch, this.twinPatchCallbackContext);`